das-cli/src/common/container_manager/vault_container_manager.py

```
@@ -0,0 +1,224 @@
+                time.sleep(interval)
+                elapsed += interval
+
+        raise DockerError("Timeout waiting for the Vault API to become ready.")
+
+    def _raise_if_container_dead(self) -> None:
+        container_name = self.get_container().name
+        try:
+            container = self.get_docker_client().containers.get(container_name)
+            container.reload()
+        except docker.errors.NotFound:
+            raise DockerError(f"Vault container {container_name} is not running.")
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)